whistleblower/targets/twitter.py

```diff
@@ -138,27 +142,73 @@ def __iter__(self):
         yield 'text', self.status.text
         yield 'document_id', self.reimbursement['document_id']
 
-    def text(self):
+    def tweet_data(self):
         """
-        Proper tweet message for the given reimbursement.
+        Proper tweet data for the given reimbursement.
         """
         profile = self.reimbursement['twitter_profile']
         if profile:
-            link = 'https://jarbas.serenata.ai/layers/#/documentId/{}'.format(
-                self.reimbursement['document_id'])
-            message = (
-                '🚨Gasto suspeito de Dep. @{} ({}). '
-                'Você pode me ajudar a verificar? '
-                '{} #SerenataDeAmor na @CamaraDeputados'
-            ).format(profile, self.reimbursement['state'], link)
-            return message
+            return self.tweet_text(), self.tweet_image()
         else:
             raise ValueError(
                 'Congressperson does not have a registered Twitter account.')
 
+
+    def tweet_text(self):
+        link = 'https://jarbas.serenata.ai/layers/#/documentId/{}'.format(
+            self.reimbursement['document_id'])
+        message = (
+            '🚨Gasto suspeito de Dep. @{} ({}). '
+            'Você pode me ajudar a verificar? '
+            '{} #SerenataDeAmor na @CamaraDeputados'
+        ).format(
+            self.reimbursement['twitter_profile'],
+            self.reimbursement['state'],
+            link
+        )
+        return message
+
+    def camara_image_url(self):
+        """
+        Proper image url for the given reimbursement.
+        """
+        url = (
+            'http://www.camara.gov.br/cota-parlamentar/documentos/publ/'
+            '{}/{}/{}.pdf'.format(
+                self.reimbursement['applicant_id'],
+                self.reimbursement['year'],
+                self.reimbursement['document_id'])
+        )
+
+        return url
+
+    def tweet_image(self):
+        """
+        Download, crop and open the image for the given reimbursement.
+        """
+        try:
+            response = urllib.request.urlopen(self.camara_image_url())
+        except urllib.error.HTTPError:
+            return None
+
+        image_bin = Image(file=response).make_blob('png')
+        numpy_array = np.frombuffer(image_bin, np.uint8)
+
+        with NamedTemporaryFile(suffix='.png') as temp:
+            crop(numpy_array, temp.name)
+
+            with open(temp.name, 'rb') as cropped_file:
+                cropped_image = cropped_file
+
+        return cropped_image
```

Comment:
Does this code work? I mean, when I exit a context manager created with `NamedTemporaryFile` I expect the file do be deleted. It might work because the `cropped_image` is still in memory, but I think this design is not neat. Does the `crop function takes a BytesIO instead? This method could return an image as bytes, for example.

Alternatively you could write your own context manager to have more control and to be more explicit about the temporary file deletion:

```python
…
from contextlib import contextmanager

…
class Tweet:

    …

    @contextmanager
    def receipt_pdf_as_png(self, pdf_response):
        image_bin = Image(file= pdf_response).make_blob('png')
        numpy_array = np.frombuffer(image_bin, np.uint8)

        # write PNG to a temp file
        temp = NamedTemporaryFile(delete=False, suffix='.png')
        with open(temp.name, 'wb') as fobj:
            crop(numpy_array, temp.name)

        # return PNG as a file object
        with open(temp.name, 'rb') as fobj:
            yield fobj

        # delete temporary file
        os.remove(temp.name)

     def tweet_image(self):
         try:
             response = urllib.request.urlopen(self.camara_image_url())
         except urllib.error.HTTPError:
             return None

        with self.receipt_pdf_as_png(response) as image:
             return image
```